### PR TITLE
xDS service discovery development document

### LINKED ISSUE #314: Added the xDS service discovery document.
### What would you like to be added?

Added the xDS service discovery document.

### Why is this needed?

 The xDS service discovery capability is added to the Sermant 2.0.0 version.

### PR DESCRIPTION
**Pull request by daizhenyu**
What type of PR is this?

/Doc

What this PR does / why we need it?

Add xDS service discovery development document

Which issue(s) this PR fixes:

Fixes #314

### CODE REVIEW

File: docs/zh/document/developer-guide/xds-discovery.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+**基于xDS协议的服务发现**允许Sermant对接Istio的控制平面获取Kubernetes service信息和其对应的具体服务实例信息。
+
+> 说明：使用Sermant的xDS服务发现能力需服务部署在Kubernetes容器环境。 
+
+## 开发示例
+
+本开发示例基于[创建首个插件](README.md)文档中创建的工程，演示插件如何通过Sermant框架提供的xDS服务发现能力获取服务实例：
+
+1. 在工程中`template\template-plugin`下的`com.huaweicloud.sermant.template.TemplateDeclarer` 类中新增变量`xdsServiceDiscovery`获取Sermant框架提供的xDS服务发现服务，用于获取服务实例：
```
Comment (lilai23) [inline docs/zh/document/developer-guide/xds-discovery.md]: package name can be the new-coming one

File: docs/zh/document/developer-guide/xds-discovery.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+         ServiceInstance instance = iterator.next();
+         System.out.println("ServiceInstance: [" + instance.getHost() + " : "
+                 + instance.getPort() + "]");
+     }
+   }
+   ```
+
+开发完成后，可参照创建首个插件时的[打包构建](README.md#打包构建)流程，在工程根目录下执行 **mvn package**后生成构建产物。
+
+   > 说明：`service-test`必须是Kubernetes service的名称，获取的服务实例信息为该service对应的pod实例数据。
```
Comment (lilai23) [inline docs/zh/document/developer-guide/xds-discovery.md]: better give a yaml example

File: docs/zh/document/developer-guide/xds-discovery.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+# 基于xDS协议的服务发现
+
+本文介绍如何在开发中使用Sermant提供的xDS服务发现能力。
+
+## 功能介绍
+
+**基于xDS协议的服务发现**允许Sermant对接Istio的控制平面获取Kubernetes service信息和其对应的具体服务实例信息。
+
+> 说明：使用Sermant的xDS服务发现能力需服务部署在Kubernetes容器环境。
```
Comment (lilai23) [inline docs/zh/document/developer-guide/xds-discovery.md]: istio is more suitable

File: docs/en/document/developer-guide/xds-discovery.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+     Iterator<ServiceInstance> iterator = serviceInstance.iterator();
+     while (iterator.hasNext()) {
+         ServiceInstance instance = iterator.next();
+         System.out.println("ServiceInstance: [" + instance.getHost() + " : "
+                 + instance.getPort() + "]");
+     }
+   }
+   ```
+
+After completing the development, refer to the [Packaging and Building](README.md#Packaging and Building) process in the Creating Your First Plugin document. Execute **mvn package** in the root directory of the project to generate the build artifact.
```
Comment (hanbingleixue) [inline docs/en/document/developer-guide/xds-discovery.md]: Format Error ： [Packaging and Building](README.md#Packaging and Building)

File: docs/en/document/developer-guide/xds-discovery.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+   > Note: `service-test` must be the name of the Kubernetes service. The obtained service instance information corresponds to the pod instance data of that service.
+
+4. Enable the xDS service and set it to be enabled in `agent/config/config.properties` with the following configuration:
```
Comment (hanbingleixue) [inline docs/en/document/developer-guide/xds-discovery.md]: The serial number should be 3.

File: docs/en/document/developer-guide/xds-discovery.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+   ```properties
+   # xDS service switch
+   agent.service.xds.service.enable=true
+   ```
+
+5. After completing the execution, package the Sermant image and the host microservice image. Start the `service-test`service in the k8s environment and create the corresponding service instances (implemented by the user). Finally, start the host application in Kubernetes and mount Sermant. Refer to the [Sermant Injector User Guide](../user-guide/sermant-injector.md#Startup and Result Verification) for guidance on packaging Sermant and the host image in the Kubernetes environment, as well as mounting Sermant for startup.
```
Comment (hanbingleixue) [inline docs/en/document/developer-guide/xds-discovery.md]: Format Error ： [Sermant Injector User Guide](../user-guide/sermant-injector.md#Startup and Result Verification)

File: docs/zh/document/developer-guide/xds-discovery.md (hunk 1 above)
Comment (daizhenyu) [reply docs/zh/document/developer-guide/xds-discovery.md]: done

File: docs/zh/document/developer-guide/xds-discovery.md (hunk 2 above)
Comment (daizhenyu) [reply docs/zh/document/developer-guide/xds-discovery.md]: done

File: docs/zh/document/developer-guide/xds-discovery.md (hunk 3 above)
Comment (daizhenyu) [reply docs/zh/document/developer-guide/xds-discovery.md]: done

File: docs/en/document/developer-guide/xds-discovery.md (hunk 4 above)
Comment (daizhenyu) [reply docs/en/document/developer-guide/xds-discovery.md]: done

File: docs/en/document/developer-guide/xds-discovery.md (hunk 5 above)
Comment (daizhenyu) [reply docs/en/document/developer-guide/xds-discovery.md]: done

**Pull request merged**